### PR TITLE
write perfmap pages as markdown

### PR DESCRIPTION
**Pull request by zyguan**
Signed-off-by: zyguan <zhongyangguan@gmail.com>

### CODE REVIEW

**Comment by cfzjywxk** [comment]
May also need to refactor the tikv part?

File: src/tidb-kv-client.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  OneOrMore(
+    Sequence(
+      NonTerminal("Fetch pending requests"),
+      Choice(
+        0,
+        Comment("tikv is not overload"),
+        NonTerminal("Fetch more requests"),
+      ),
+      NonTerminal("Select a connection"),
+      NonTerminal("Build a batch request and send it"),
```
Comment (you06) [inline src/tidb-kv-client.md:62]: I think this step only appends the batch request into the control buffer of the streaming connection and another goroutine will send it asynchronously. So there should be a missing latency after this step.

File: src/tidb-kv-client.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ),
+  ),
+)
+```
+
+- Duraion of each iteration is obeserved as `tidb_tikvclient_batch_send_latency` (exclude waiting for the first request).
+- If the target TiKV is overload, more requests may be collected for sending. The event is only counted by `tidb_tikvclient_batch_wait_overload` (without waiting duration).
+- The connection (*batchCommandsClient*) is chosen round-robinly, we try to acquire a lock before using the connection. *no available connections* might be reported if we cannot find such a connection, such an event is counted by `tidb_tikvclient_batch_client_no_available_connection_total`.
```
Comment (you06) [inline src/tidb-kv-client.md:70]: I'm considering whether there is a more understandable way for metrics description, more direct correspondence between metrics and diagrams, e.g. highlight when metrics or diagrams are hovered.

**Comment by cfzjywxk** [comment]
Also merging this pr we could add a preview entry in the repo describtion.

File: src/tidb-kv-client.md (hunk 1 above)
Comment (zyguan) [reply src/tidb-kv-client.md:62]: The diagram describes the [batch send loop](https://github.com/tikv/client-go/blob/be31f33ba03b543bb20503c3264877e09440d30c/internal/client/client_batch.go#L299) and this step is corresponding to [here](https://github.com/tikv/client-go/blob/be31f33ba03b543bb20503c3264877e09440d30c/internal/client/client_batch.go#L374-L385). Do you mean https://github.com/you06/performance-map/pull/3/files#diff-8354e42366dea0e62bb1095703d36c40ada5924ddd9c1d817ff2e5736b7ededbR26-R27 ?

**Comment by you06** [comment]
> Also after merging this pr we could add a preview entry in the repo describtion.

Yes, need to update the build workflow

File: src/tidb-kv-client.md (hunk 2 above)
Comment (zyguan) [reply src/tidb-kv-client.md:70]: It seems many spans described in diagrams do not have corresponding metrics by now. We may need to refine metrics firstly. The UI is not too hard.

**Comment by zyguan** [comment]
> May also need to refactor the tikv part?

Yes

**Comment by zyguan** [comment]
> > Also after merging this pr we could add a preview entry in the repo describtion.
> 
> Yes, need to update the build workflow

We may need to add an index page.

**Comment by you06** [comment]
> We may need to add an index page.

Can `tidb-query-duration` be renamed to `index`?

File: src/tidb-kv-client.md (hunk 1 above)
Comment (you06) [reply src/tidb-kv-client.md:174]: What I mentioned is the mechanism inside gRPC, it has a [buffer](https://github.com/grpc/grpc-go/blob/71f16a1d4afab21dea1732c89401d2e7b9949d61/internal/transport/http2_client.go#L79-L81) for messages to be sent. The client just [put the messages into the buffer](https://github.com/grpc/grpc-go/blob/71f16a1d4afab21dea1732c89401d2e7b9949d61/internal/transport/http2_client.go#L974) by calling `Send`.

**Comment by zyguan** [comment]
> > We may need to add an index page.
> 
> Can `tidb-query-duration` be renamed to `index`?

Oh, I just add a seperate index page.

File: src/tidb-kv-client.md (hunk 1 above)
Comment (zyguan) [reply src/tidb-kv-client.md:174]: I see, how can we observe such a duration?

Comment (you06) [reply src/tidb-kv-client.md:174]: It's unable to observe such duration unless we can add some metrics inside gRPC. Let's left a grey block for the unclear part.

**Comment by cfzjywxk** [comment]
@you06 
PTAL

File: src/tidb-kv-client.md (hunk 1 above)
Comment (you06) [reply src/tidb-kv-client.md:174]: What about

```
NonTerminal("Build a batch request and append it to the sending buffer"),
NonTerminal("Send asynchronously"),
```

The first section is under client-go's control and the second section can be related to the utilization of the system.

**Review by you06 — APPROVED** [review]

**Pull request merged**